lib/streamlit/navigation/page.py

```diff
@@ -114,17 +122,26 @@ def __init__(
         inferred_icon = ""
         if isinstance(page, Path):
             inferred_icon, inferred_name = page_icon_and_name(page)
+        elif hasattr(page, "__name__"):
+            inferred_name = str(page.__name__)
         elif title is None:
-            if hasattr(page, "__name__"):
-                inferred_name = str(page.__name__)
-            else:
-                raise StreamlitAPIException(
-                    "Cannot infer page title for Callable. Set the `title=` keyword argument."
-                )
+            raise StreamlitAPIException(
```

Comment:
nit: it might be easier to understand the exception when there is a brief comment, e.g. `# if its not a page, does not have a name and also no title, it is a callable.`